### PR TITLE
feat: socket activated c8y-remote-access-plugin service

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Use systemd socket activated service to spawn the c8y-remote-access-plugin connection so that it is independent from the tedge-mapper-c8y process which allows the tedge-mapper-c8y service to be restarted without interrupting the connection (or performing actions like `tedge reconnect c8y`).

The c8y_RemoteAccessConnect operation will launch the connection using the socket only if is available, so it should make it flexible for OS's with and without systemd.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

* PR is current rebase upon https://github.com/thin-edge/thin-edge.io/pull/3006
* https://github.com/thin-edge/thin-edge.io/issues/2859

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: configuration/package_scripts/_generated/c8y-remote-access-plugin/apk/postinst [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/bin/sh
 set -e
 
+
+
+
```
Comment (reubenmiller) [inline configuration/package_scripts/_generated/c8y-remote-access-plugin/apk/postinst:6]: Note, these blank sections are a result of a non-ideal implementation in the "generate.py" script which does not skip empty evaluated template blocks. This behaviour is not different to before, and other `postinst` scripts (mostly for apk packages) also have the same "empty scripts". This should be fixed, but it is not part of this PR.

**Review by rina23q — APPROVED** [review]

**Comment by rina23q** [comment]
Approved for changes except Rust part (as I contributed to the Rust part, I shouldn't approve the part)

**Review by jarhodes314 — APPROVED** [review]
I'm happy with the Rust parts

**Review by gligorisaev — COMMENTED** [review]
The created tests are doing what they are created for, no objections from my side

File: plugins/c8y_remote_access_plugin/src/input.rs [hunk 2]
```diff
@@ -79,7 +82,13 @@ impl TryFrom<C8yRemoteAccessPluginOpt> for Command {
             C8yRemoteAccessPluginOpt {
                 connect_string: Some(message),
                 ..
-            } => Ok(Command::SpawnChild(message)),
+            } => {
+                if Path::new(UNIX_SOCKFILE).exists() {
```
Comment (Bravo555) [inline plugins/c8y_remote_access_plugin/src/input.rs:86]: The test `input::tests::parses_spawn_child_if_no_flag_is_provided` just stopped working for me because of this branch:

```
failures:

---- input::tests::parses_spawn_child_if_no_flag_is_provided stdout ----
thread 'input::tests::parses_spawn_child_if_no_flag_is_provided' panicked at plugins/c8y_remote_access_plugin/src/input.rs:164:9:
assertion `left == right` failed
  left: TryConnectUnixSocket("530,jrh-rc-test0,127.0.0.1,22,cd8fc847-f4f2-4712-8dd7-31496aef0a7d")
 right: SpawnChild("530,jrh-rc-test0,127.0.0.1,22,cd8fc847-f4f2-4712-8dd7-31496aef0a7d")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    input::tests::parses_spawn_child_if_no_flag_is_provided
```

I have `/run/c8y-remote-access-plugin.sock` file in my filesystem, and I don't know if it should be there or if some other test perhaps created it.

Also I think it's a bit unexpected for `TryFrom` impl to be doing side effects.

Comment (rina23q) [reply plugins/c8y_remote_access_plugin/src/input.rs:86]: Exactly because you have `/run/c8y-remote-access-plugin.sock` file in the device that is running test, it failed. The test should be modified to pass both case.

Comment (rina23q) [reply plugins/c8y_remote_access_plugin/src/input.rs:86]: Fixed by #3041
